maintenance_request_time_performance/models/maintenance.py

```diff
@@ -0,0 +1,65 @@
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class MaintenanceStage(models.Model):
+    _inherit = 'maintenance.stage'
+
+    in_progress = fields.Boolean(
+        'In Progress',
+        default=False)
+
+    @api.model
+    def update_default_stage_in_progress(self):
+        stage = self.env.ref('maintenance.stage_1')
+        if stage:
+            stage.write({'in_progress': True})
+
+    @api.constrains('in_progress', 'done')
+    def check_maintenance_stage(self):
+        if self.in_progress and self.done:
+            raise ValidationError(
+                _("It can´t be in progress and done at the same time"))
+
+
+class MaintenanceRequest(models.Model):
+    _inherit = 'maintenance.request'
+
+    start_task_datetime = fields.Datetime(
+        string='Start Date and time of maintenance task completion')
+    end_task_datetime = fields.Datetime(
+        string='End Date and time of maintenance task completion')
+    real_task_duration = fields.Float(
+        compute='_compute_real_task_duration',
+        string='Real Duration',
+        store=True)
+
```

Comment:
a computed, stored field name `duration_difference` can be added and used in pivot view